driver/src/main/scala/api/indexes.scala

```diff
@@ -311,26 +463,33 @@ object IndexesManager {
     }
   }
 
-  implicit object NSIndexReader extends BSONDocumentReader[NSIndex] {
```

Comment:
Split the BSON reader with one for `NSIndex` and one for `Index`, useful to read only `Index` for manager working on a specific collection